R/map_fetch.R

```diff
@@ -0,0 +1,268 @@
+#' 	Defaults to classic.point.
+#' 	
+#' 	@param search (optional): defines what type of subset of all GBIF data to 
+#' 	return. Should be one of c("taxonKey", "datasetKey", "country", "publisher",
+#' 	"publishingCountry"). Without any search parameter, all occurrences will be 
+#' 	returned.
+#' 	
+#' 	@param id (optional): defines the value to be used as filter criterium in 
+#' 	the category supplied by (\code{search}). Appropriate values depend on the
+#' 	search category that is used, for example integer for 
+#' 	(\code{search = "taxonKey"}). Has to be provided if 
+#' 	(\code{search}) parameter is specified.
+#' 	
+#' 	@param year (optional): limits the occurrences within a range of years, 
+#' 	for example c(2008, 2009, 2010).
+#' 	
+#' 	@param basisOfRecord (optional): can be given multiple times to include 
+#' 	records with that basis of record. The full list is: c("OBSERVATION", 
+#' 	"HUMAN_OBSERVATION", "MACHINE_OBSERVATION", "MATERIAL_SAMPLE", 
+#' 	"PRESERVED_SPECIMEN", "FOSSIL_SPECIMEN", "LIVING_SPECIMEN", 
+#' 	"LITERATURE", "UNKNOWN".
+#' 	
+#' @return ###???
+#'
+#' @details This function uses the arguments passed on to generate a query
+#' to the GBIF web map API. The API returns a web tile object as png that can be
+#' read and converted into an R raster object. The break values or nbreaks
+#' generate a custom colour palette for the web tile, with each bin
+#' corresponding to one grey value. After retrieval, the raster is reclassified
+#' to the actual break values. This is a somewhat hacky but nonetheless
+#' functional solution in the absence of a GBIF raster API implementation.
+#'
+#' @author Laurens Geffert, \email{laurensgeffert@@gmail.com}
+#' @references \url{https://www.gbif.org/developer/maps}
+#' @keywords web map, web tile, GBIF
+#' @examples \dontrun{}
+
+map_fetch <- function(
+  source = 'density',
+  x = 0,
+  y = 0,
+  z = 0,
+  format = '@1x.png',
+  srs = 'EPSG:4326',
+  bin = NULL,
+  hexPerTile = NULL,
+  squareSize = NULL,
+  style = 'classic.point',
+  search = NULL,
+  id = NULL,
+  year = NULL,
+  basisOfRecord = NULL
+  ) {
+  
+  
+  # Check input ----------------------------------------------------------------
+  source <- match.arg(
+    arg = source,
+    choices = c('density', 'adhoc'),
+    several.ok = FALSE
+  )
+
+  if(!is.numeric(x) & !is.integer(x)){
+    stop('x value should be numeric or integer.', call. = FALSE)
+  }
+  
+  if(!is.numeric(y) & !is.integer(y)){
+    stop('y value should be numeric or integer.', call. = FALSE)
+  }
+  
+  if(!is.numeric(z) & !is.integer(z)){
+    stop('z value should be numeric or integer.', call. = FALSE)
+  }
+
+  format <- match.arg(
+    arg = format,
+    choices = c('.mvt', '@Hx.png', '@1x.png', '@2x.png', '@3x.png', '@4x.png'),
+    several.ok = FALSE
+  )
+  
+  z <- match.arg(
+    arg = z,
+    choices = c('EPSG:3857', 'EPSG:4326', 'EPSG:3575', 'EPSG:3031'),

```

Comment:
seems like these `EPSG` strings are for Spatial reference system, no?  isn't `z` the zoom level?

Reply:
oups, that's a bug. Thanks for spotting it!